unity-client/Assets/Scripts/MainScripts/DCL/Controllers/Scene/MessagingBus.cs

```diff
@@ -183,13 +181,13 @@ public void Enqueue(QueuedSceneMessage message, QueueMode queueMode = QueueMode.
 
                 if (type == MessagingBusType.INIT)
                 {
-                    manager.pendingInitMessagesCount++;
+                    Environment.i.messagingControllersManager.pendingInitMessagesCount++;
```

Comment:
Please keep the `manager` reference. This is cached for a reason, this is a performance critical code and believe it or not, the `i` getter was consuming a few ms in high stress cases.

Reply:
You were right! I've made some performance test and then decided to change some structures here.

Reply:
After making some changes, now the non-local access is as fast as the local one. Please see the changes in the static variable in Environment, the removal of the accessor for the messagingControllerManager member, and also the changes made to Singleton.